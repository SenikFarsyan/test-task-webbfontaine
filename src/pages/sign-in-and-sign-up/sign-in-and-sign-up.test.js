import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import SignInAndSignUpPage from './sign-in-and-sign-up.component';

const mockStore = configureStore([]);

describe('connected react-redux SignInAndSignUpPage component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      user: {
        messages: {}
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <SignInAndSignUpPage />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
