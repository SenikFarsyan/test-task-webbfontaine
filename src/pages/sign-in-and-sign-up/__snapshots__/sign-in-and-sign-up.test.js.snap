// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected react-redux SignInAndSignUpPage component should render with given state from Redux store 1`] = `
<div
  className="signinAndSignup"
>
  <h3>
    Sign in and Sign up page
  </h3>
  <form
    className="form"
  >
    <input
      onChange={[Function]}
      placeholder="name"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
      value=""
    />
    <div>
      <input
        name="register"
        onClick={[Function]}
        type="button"
        value="Register"
      />
      <input
        name="signin"
        onClick={[Function]}
        type="button"
        value="Signin"
      />
    </div>
  </form>
  <div
    className="successMessage "
  >
    <p />
    <p />
  </div>
</div>
`;
